xcm/src/v0/multi_location.rs

```diff
@@ -44,77 +44,423 @@ use parity_scale_codec::{self, Decode, Encode};
 /// This specific `MultiLocation` implementation uses a Rust `enum` in order to make pattern matching easier.
 ///
 /// The `MultiLocation` value of `Null` simply refers to the interpreting consensus system.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug)]
-pub enum MultiLocation {
-	/// The interpreting consensus system.
-	Null,
-	/// A relative path comprising 1 junction.
-	X1(Junction),
-	/// A relative path comprising 2 junctions.
-	X2(Junction, Junction),
-	/// A relative path comprising 3 junctions.
-	X3(Junction, Junction, Junction),
-	/// A relative path comprising 4 junctions.
-	X4(Junction, Junction, Junction, Junction),
-	/// A relative path comprising 5 junctions.
-	X5(Junction, Junction, Junction, Junction, Junction),
-	/// A relative path comprising 6 junctions.
-	X6(Junction, Junction, Junction, Junction, Junction, Junction),
-	/// A relative path comprising 7 junctions.
-	X7(Junction, Junction, Junction, Junction, Junction, Junction, Junction),
-	/// A relative path comprising 8 junctions.
-	X8(Junction, Junction, Junction, Junction, Junction, Junction, Junction, Junction),
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub struct MultiLocation {
+	parents: u8,
```

Comment:
not quite clear to me (yet) how parents should be interpreted, maybe add some docs for it?

Reply:
(reading the docs of `parent_count` clarified it, but maybe write a line here as well)